tracing/thrift.go

```diff
@@ -3,116 +3,145 @@ package tracing
 import (
 	"context"
 	"fmt"
+	"math/rand"
 	"strconv"
 
-	"github.com/reddit/baseplate.go/log"
 	"github.com/reddit/baseplate.go/set"
 	"github.com/reddit/baseplate.go/thriftbp"
 
 	"github.com/apache/thrift/lib/go/thrift"
-	zipkin "github.com/openzipkin/zipkin-go"
-	"github.com/openzipkin/zipkin-go/model"
 )
 
-// StartSpanFromThriftContext creates a span from the thrift context object.
+// StartSpanFromThriftContext creates a server span from thrift context object.
 //
 // This span would usually be used as the span of the whole thrift endpoint
-// handler, and the parent of the subspans.
+// handler, and the parent of the child-spans.
 //
 // Caller should pass in the context object they got from thrift library,
 // which would have all the required headers already injected.
 //
 // Please note that "Sampled" header is default to false according to baseplate
 // spec, so if the context object doesn't have headers injected correctly,
-// this span (and all its subspans) will never be sampled.
+// this span (and all its child-spans) will never be sampled,
+// unless debug flag was set later.
 //
 // If any of the tracing related thrift header is present but malformed,
 // it will be ignored.
-// The error will also logged if the given logger is non-nil.
+// The error will also logged if the global tracer's logger is non-nil.
 // Absent tracing related headers are always silently ignored.
-func StartSpanFromThriftContext(ctx context.Context, optName string, logger log.Wrapper) zipkin.Span {
-	var parentCtx model.SpanContext
+func StartSpanFromThriftContext(ctx context.Context, name string) *Span {
+	return StartSpanFromThriftContextWithTracer(ctx, name, nil)
```

Comment:
`newSpan` handles that: https://github.com/reddit/baseplate.go/blob/333027d81e000254eefa7362d6ad9dc59a92e670/tracing/span.go#L54-L56

But this code does have a bug to panic because we tried to access `tracer.Logger`. Fixed.